source/loader/ur_lib.cpp

```diff
@@ -215,6 +215,17 @@ urLoaderConfigSetCodeLocationCallback(ur_loader_config_handle_t hLoaderConfig,
     return UR_RESULT_SUCCESS;
 }
 
+ur_result_t
+urLoaderConfigSetLoggerCallback(ur_logger_output_callback_t pfnLoggerCallback) {
+    if (!pfnLoggerCallback) {
+        return UR_RESULT_ERROR_INVALID_NULL_POINTER;
+    }
+
+    logger::get_logger().setCallbackSinkFunction(pfnLoggerCallback);
```

Comment:
I meant to highlight this as a question because I wasn't sure about how to get the callback function pointer into the logger as `get_logger` takes a name parameter. I can't see any loader references in DDI tables, would I need to add one for the loader or should the entry point be part of another group maybe?

Reply:
This is not really a loader API, so I'd call this something like `urSetLoggerCallback` and remove `loader_only` from the spec. This will generate a DDI function where you can implement this for each adapter. And each adapter will then need to have the `logger::get_logger().setSink()` call in their implementation. Because the instance returned by `logger::get_logger()` is different for each adapter.

Additionally, if you want the application to be able to also get callbacks on loader-related messages, you'd need to make an exception in the template file so that do the `logger::get_logger().setSink()` at the loader layer, before the function is forwarded to the adapter-specific implementation.